Document readline keybindings (#31256)
nodejs/node: doc/api/repl.md

  variables. When pressed while entering other input, displays relevant
  autocompletion options.

> For key combinations to Reverse-i-search, refer to [`Reverse-i-search`][].

Reviewer: ```suggestion
For key bindings related to the reverse-i-search, see [`reverse-i-search`][].
```

Author: @BridgeAR , ditto.

> For a full list of special keys, refer to [TTY keybindings][].

Reviewer: ```suggestion
For all other key bindings, see [TTY keybindings][].
```

Author: @BridgeAR , thanks. Fixed it.

[`repl.ReplServer`]: #repl_class_replserver
[`repl.start()`]: #repl_repl_start_options
[`util.inspect()`]: util.html#util_util_inspect_object_options

> [`Reverse-i-search`]: #repl_Reverse_i_search

Reviewer: ```suggestion
[`reverse-i-search`]: #repl_reverse_i_search
```

Author: @BridgeAR , ditto,
